File: examples/governance_static_go/domain/regulatory_framework.go
package domain

import (
	"time"
	"github.com/google/uuid"
)

type RegulatoryFramework struct {
	ID        string    `json:"id"`
	Name string `json:"name"`
	Jurisdiction string `json:"jurisdiction"`
	Version string `json:"version"`
	EffectiveDate time.Time `json:"effective_date"`
	Authority string `json:"authority"`
	Requirements []FrameworkRequirement `json:"requirements"`
	Status string `json:"status"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

func NewRegulatoryFramework(name string, jurisdiction string, version string, effectiveDate time.Time, authority string, requirements []FrameworkRequirement, status string) *RegulatoryFramework {
	a := &RegulatoryFramework{
		ID:        uuid.New().String(),
		Name: name,
		Jurisdiction: jurisdiction,
		Version: version,
		EffectiveDate: effectiveDate,
		Authority: authority,
		Requirements: requirements,
		Status: status,
		CreatedAt: time.Now(),
		UpdatedAt: time.Now(),
	}
	return a
}

func (a *RegulatoryFramework) Validate() error {
	if a.Name == "" {
		return &ValidationError{Field: "name", Message: "name can't be blank"}
	}
	if a.Jurisdiction == "" {
		return &ValidationError{Field: "jurisdiction", Message: "jurisdiction can't be blank"}
	}
	return nil
}

File: examples/governance/compliance_domain/lib/compliance_domain/regulatory_framework/regulatory_framework.rb
require 'hecks/mixins/model'

module ComplianceDomain
  class RegulatoryFramework
    autoload :FrameworkRequirement, "compliance_domain/regulatory_framework/framework_requirement"
    autoload :Lifecycle, "compliance_domain/regulatory_framework/lifecycle"

    include Hecks::Model

    attribute :name
    attribute :jurisdiction
    attribute :version
    attribute :effective_date
    attribute :authority
    attribute :requirements, default: [], freeze: true
    attribute :status

    # State predicates — see lifecycle.rb for full state machine
    def draft?; status == "draft"; end
    def active?; status == "active"; end
    def retired?; status == "retired"; end

    private

    def validate!
      raise ValidationError.new("name can't be blank", field: :name, rule: :presence) if name.nil? || (name.respond_to?(:empty?) && name.empty?)
      raise ValidationError.new("jurisdiction can't be blank", field: :jurisdiction, rule: :presence) if jurisdiction.nil? || (jurisdiction.respond_to?(:empty?) && jurisdiction.empty?)
    end
  end
end
